webook/arrangement/forms/person_forms.py

```diff
@@ -0,0 +1,18 @@
+from django import forms
+from django.db.models import Q
+
+from webook.arrangement.models import Arrangement, Event, Person
+from webook.users.models import User
+
+
+class AssociatePersonWithUserForm(forms.ModelForm):
+    user_set = forms.ModelChoiceField(queryset = User.objects.filter( Q(person__isnull=True) ))
```

Comment:
Is this tested? does it work? I would have guessed filter without `Q` and maybe `exclude` to remove any empty strings would suffice. 
Why does this work is more what im wondering about.

Reply:
The Q() is redundant here, and is a mistake on my part. I have removed it.

As for the intent here it is to get all users that do not have a person associated with them. We know that if user.person is null == True then that user does not have a person, hence we want to present that user as a selectable choice when selecting a user for a person. If a user already has a person then that user is not of interest in this process. 
I don't think this utility in its current form will be used extensively by anyone but developers or admins, and it may have to be revised or condensed come the feature set from the SSO branch. Which is why I elected to do things as simple as possible.

Not sure why we would do any excludes, I think the query as is suffices. ModelChoiceField is really magical -- all we're really doing here is giving it a queryset (which is the result of our filtering operation, a queryset containing all users without any associated people), and the rest is handled by Django. There's no handling of strings or anything, in the template we just go ``{{ form.user_set }}` and it is rendered as a multiselect, with the users in the resulting queryset displayed and selectable. (I did some more custom stuff in the template, but in essence it's as easy as that)

Reply:
yes, if an empty string is not an option, then no need to use it. 
its very common to see `filter(...).exclude(...)` in tandem, the filter will get all that is null, but since empy string is falsy in python you will have to exclude any empty values to get the real `NUL` values from the query. 

thanks for the explaination, I now understand what your doing.